pir/cpp/client_test.cpp

```diff
@@ -72,12 +77,50 @@ TEST_F(PIRClientTest, TestProcessResponse) {
   Context()->Encoder()->encode(value, pt);
   vector<Ciphertext> ct(1);
   Encryptor()->encrypt(pt, ct[0]);
-  PIRPayload payload = PIRPayload::Load(ct);
 
-  auto result = client_->ProcessResponse(payload).ValueOrDie();
+  Response reply;
+  SaveCiphertexts(ct, reply.mutable_reply());
+
+  auto result = client_->ProcessResponse(reply).ValueOrDie();
   ASSERT_EQ(result, value);
 }
 
+TEST_F(PIRClientTest, TestRequestSerialization) {
+  int64_t value = 987654321;
+  Plaintext pt;
+  Context()->Encoder()->encode(value, pt);
+  vector<Ciphertext> ct(1);
+  Encryptor()->encrypt(pt, ct[0]);
+
+  Response reply_proto;
+  SaveCiphertexts(ct, reply_proto.mutable_reply());
+
+  auto reloaded = LoadCiphertexts(Context()->SEALContext(), reply_proto.reply())
+                      .ValueOrDie();
```

Comment:
Rather than doing ValueOrDie, it's a lot easier to debug the test if you do an assert on status being ok and output the status to_string value.